query-compiler/query-compiler/tests/snapshots/queries__queries@create-m2m.json.snap

```diff
@@ -0,0 +1,38 @@
+---
+source: query-compiler/query-compiler/tests/queries.rs
+expression: pretty
+input_file: query-compiler/query-compiler/tests/data/create-m2m.json
+---
+let 0 = unique (query «INSERT INTO "public"."User" ("email") VALUES ($1)
+                       RETURNING "public"."User"."id"»
+                params [const(String("user.1737556027349@prisma.io"))])
+in let 0$id = mapField id (get 0)
+   in let 1 = unique (query «INSERT INTO "public"."Post" ("title","userId")
+                             VALUES ($1,$2) RETURNING "public"."Post"."id"»
+                      params [const(String("How to make an omelette")),
+                              var(0$id as Int)])
+      in let 2 = unique (query «INSERT INTO "public"."Category" ("name") VALUES
+                                ($1) RETURNING "public"."Category"."id"»
+                         params [const(String("cooking"))])
+         in let 1$id = mapField id (get 1);
+                2$id = mapField id (get 2)
+            in execute «INSERT INTO "public"."_CategoryToPost" ("B","A") VALUES
+                        ($1,$2) ON CONFLICT DO NOTHING»
+               params [var(1$id as Int), var(2$id as Int)];
+   let 4 = let 0$id = mapField id (get 0)
```

Comment:
[unrelated to this PR] We should later introduce optimization passes and, among them, eliminate duplicate `mapField` expressions (here we even already have it in scope). Although they are cheap in this case (since `0` is `unique`), in general we may be mapping a long list.